error tracking costs docs (#11385)
PostHog/posthog.com: contents/docs/error-tracking/cutting-costs.mdx

---
title: Cutting error tracking costs
sidebar: Docs
showTitle: true
---

We aim to be significantly cheaper than our competitors. In addition to our [ridiculously cheap pricing](/pricing), below are tips to reduce your error tracking costs:

## Configure exception autocapture

> Exception autocapture is a powerful feature that captures every exception your users encounter. It can also capture more than you need. To reduce which exceptions that are captured, you can configure which types of exceptions are autocaptured in the JS SDK config. By default we capture all unhandled errors and rejections.

Reviewer: ```suggestion
By default, we capture all unhandled errors and rejections. This can capture more than you need. To reduce which exceptions that are captured, you can configure which types of exceptions are autocaptured in the JS SDK config like this: 
```

```json
{
  "capture_exceptions": {
    "capture_unhandled_errors": true,
    "capture_unhandled_rejections": true,
    "capture_console_errors": false
  }
}
```

Alternatively, you can disable exception autocapture completely in your [project settings](https://us.posthog.com/error_tracking/configuration).

## Filtering exceptions clientside

The [`before_send`](/docs/libraries/js/features#amending-or-sampling-events) callback in the JS SDK will drop events if a falsey value is returned. You can provide a `before_send` function when initializing the PostHog client to exclude any exception events you do not wish to capture client-side.

```js
posthog.init('<ph_project_api_key>', {
    before_send: (event) => {
        if (event.event === "$exception") {
            const exceptionList = event.properties["$exception_list"] || []
            const exception = exceptionList.length > 0 ? exceptionList[0] : null;

            if (exception && exception["$exception_type"] == "UnwantedError") {
                return false
            }
        }
        return event
    }
})
```

## Issue suppression

> If the exceptions you do not wish to capture are all being grouped under the same issue it is possible to suppress them so that subsequent exception events will not be ingested. Issues can be suppressed from the list or issue page by changing their status to Suppressed.

Reviewer: ```suggestion
If the exceptions you do not wish to capture are all being grouped under the same issue, you can suppress them so that subsequent exception events will not be ingested. Issues can be suppressed from the list or issue page by changing their status to **Suppressed**.
```
